payments-core/src/test/java/com/stripe/android/DefaultIntentConfirmationInterceptorTest.kt

```diff
@@ -0,0 +1,365 @@
+import org.junit.Test
+import org.mockito.kotlin.mock
+
+class DefaultIntentConfirmationInterceptorTest {
+
+    @Test
+    fun `Returns confirm as next step if invoked with client secret for existing payment method`() = runTest {
+        val interceptor = DefaultIntentConfirmationInterceptor(
+            stripeRepository = object : AbsFakeStripeRepository() {},
+            publishableKeyProvider = { "pk" },
+            stripeAccountIdProvider = { null },
+            createIntentCallback = null,
+        )
+
+        val paymentMethod = PaymentMethodFixtures.CARD_PAYMENT_METHOD
+
+        val nextStep = interceptor.intercept(
+            clientSecret = "pi_1234_secret_4321",
+            paymentMethod = paymentMethod,
+            shippingValues = null,
+            setupForFutureUsage = null,
+        )
+
+        val confirmNextStep = nextStep as? IntentConfirmationInterceptor.NextStep.Confirm
+        val confirmParams = confirmNextStep?.confirmParams as? ConfirmPaymentIntentParams
+
+        assertThat(confirmParams?.paymentMethodId).isEqualTo(paymentMethod.id)
+        assertThat(confirmParams?.paymentMethodCreateParams).isNull()
+    }
+
+    @Test
+    fun `Returns confirm as next step if invoked with client secret for new payment method`() = runTest {
+        val interceptor = DefaultIntentConfirmationInterceptor(
+            stripeRepository = object : AbsFakeStripeRepository() {},
+            publishableKeyProvider = { "pk" },
+            stripeAccountIdProvider = { null },
+            createIntentCallback = null,
+        )
+
+        val createParams = PaymentMethodCreateParamsFixtures.DEFAULT_CARD
+
+        val nextStep = interceptor.intercept(
+            clientSecret = "pi_1234_secret_4321",
+            paymentMethodCreateParams = createParams,
+            shippingValues = null,
+            setupForFutureUsage = null,
+        )
+
+        val confirmNextStep = nextStep as? IntentConfirmationInterceptor.NextStep.Confirm
+        val confirmParams = confirmNextStep?.confirmParams as? ConfirmPaymentIntentParams
+
+        assertThat(confirmParams?.paymentMethodId).isNull()
+        assertThat(confirmParams?.paymentMethodCreateParams).isEqualTo(createParams)
+    }
+
+    @Test
+    fun `Fails if invoked without a confirm callback for existing payment method`() = runTest {
+        val interceptor = DefaultIntentConfirmationInterceptor(
+            stripeRepository = object : AbsFakeStripeRepository() {},
+            publishableKeyProvider = { "pk" },
+            stripeAccountIdProvider = { null },
+            createIntentCallback = null,
+        )
+
+        val nextStep = interceptor.intercept(
+            clientSecret = null,
+            paymentMethod = PaymentMethodFixtures.CARD_PAYMENT_METHOD,
+            shippingValues = null,
+            setupForFutureUsage = null,
+        )
+
+        assertThat(nextStep).isInstanceOf(IntentConfirmationInterceptor.NextStep.Fail::class.java)
```

Comment:
What error are we exposing for this? It would be nice to have a clear developer facing message here. I assume with our current public API shape, this is going to occur for developers when initially writing the itegration.